restli-tools/src/main/resources/apiVmTemplates/rest.delete.vm

```diff
@@ -15,70 +15,112 @@
 *#
 #if($is_interface)
   #foreach($withEG in [true, false])
```

Comment:
If we switch the order (do withEg and flattenAssocKey outside the is_interface check), then the common logic to compute `keyStubNoOptional` `keyStubWithOptional` etc (lines 18 - 45) can be de duplicated.

foreach withEg
  foreach flattenAssocKey
   // Common logic: showTemplate, keyStubNoOptional, keyStubWithOptional
   if is_interface
      // interface
   else
      // impl

Reply:
Not really. We defined some class in `is_interface`.

Also I intend to use "is_interface" in the top level. Because initially we really want to use separate files for interface but later I choose to keep them in the same file. That is a deliberate decision.

Reply:
Extracting the logic for the different method variations like EG and flattenAssocKey outside doesn't conflict with the points you mentioned.

If you think about it, the method variants are common between interface and impl.

Reply:
I undertsand you are saying the "showTeplate" is redundant. But  according to you, the interface checking will be redundant. 
```
if is_interface
// define some class
endif


foreach withEg
foreach flattenAssocKey
// Common logic: showTemplate, keyStubNoOptional, keyStubWithOptional
if is_interface
// interface
else
// impl
endforeach
endforeach
```

My preference is to keep some duplicates right now, so the interface/impl can be two separate logic groups.

Reply:
```
foreach withEg
  foreach flattenAssocKey
    // Common logic: showTemplate, keyStubNoOptional, keyStubWithOptional
    if is_interface
      // interface
      // define some class
    else
      // impl
    endif
  endforeach
endforeach
```

This is what I was proposing. You are still keeping interface/impl as two logic groups, while also de-duplicating the common logic between them.

Reply:
in your code, `define some class` needs guarding variables (which will make the code messy), otherwise there will be multiple definition of same class. That is the biggest problem.

Plus if we put "is_interface" inside foreach withEg, than better change every file to make it consitent.

Reply:
Okay, got it. Thanks for the clarification. We can keep it as it is now.